pkg/istio-agent/agent.go

```diff
@@ -329,6 +335,9 @@ func (sa *Agent) FindRootCAForXDS() string {
 		// This was never completely correct - PROV_CERT are only intended for auth with CA_ADDR,
 		// and should not be involved in determining the root CA.
 		return sa.secOpts.ProvCert + "/root-cert.pem"
+	} else if sa.secOpts.FileMountedCerts {
```

Comment:
is FileMountedCerts ever set when MetadataClientRootCert is not? let me look

Reply:
We do not have that validation in-place but it never happens

Reply:
are you thinking it would be safe to check existence of that certs vs isFileMountedcerts? IF yes, I am OK moving to that check

Reply:
My main concern is if somehow this value is set but then the root cert path wasn't. That seems like it's probably a misconfiguration but possibly one that's easy to misconfigured? So maybe we just need to have some log or failure in that case

Reply:
`getRootCertificate` already errors out if the file determined  by FindRootCAForXDS does not exists. That should cover this case?